app/src/main/java/com/github/onlynotesswent/model/note/Note.kt

```diff
@@ -60,101 +60,87 @@ data class Note(
       val publicPath: String
   )
 
-  /**
-   * Represents a list of Comments for a Note. The class is immutable. It's companion object is used
-   * to produce a new CommentCollection to get a new one.
-   */
+  /** Represents a list of Comments for a Note. The class is immutable. */
   class CommentCollection(val commentsList: List<Comment> = emptyList()) {
+    /**
+     * Retrieves a list of comments made by a specific user.
+     *
+     * @param userId The unique identifier of the user whose comments are to be retrieved.
+     * @return A list of [Comment]s associated with the specified user ID. If the user has not made
+     *   any comments, an empty list is returned.
+     */
     fun getUserComments(userId: String): List<Comment> {
       return commentsList.filter { it.userId == userId }
     }
 
-    companion object {
+    /**
+     * Adds a new comment to the CommentCollection with a unique ID.
+     *
+     * @param userId The ID of the user adding the comment.
+     * @param userName The username of the added comment user.
+     * @param content The text content of the comment.
+     * @return An updated CommentCollection including the new comment.
+     */
+    fun addComment(userId: String, userName: String, content: String): CommentCollection {
```

Comment:
Maybe this function can be renamed to addCommentByUser to have a more specific name